### DISCUSSION TITLE
Is there an example that shows using Firebase and Metamask as options to log in?

### DISCUSSION THREAD

**Discussion by obversepebble** [Web3Auth/Web3Auth#1144] [Help]
Hi, is there an example that shows using Firebase and Metamask as options to log in in the same application? Firebase provides non-wallet options to signing in (I see it in the Firebase example as expected), but if the user chooses to connect with Metamask or Walletconnect, I'd like to set the authenticated user in Firebase as well, e.g. Moralis does that here: https://docs.moralis.io/docs/web3-firebase-authentication, but I'd like to achieve the same without Moralis. Help is much appreciated.

**Comment by shahbaz17** [comment]
Hey @obversepebble 

Do you mean two options to log in? One is Firebase, like https://main.d2259cfl2oapbx.amplifyapp.com, and should this also have the option to log in with MetaMask or Phantom?

**Reply by shahbaz17** [reply]
If so, then yes, it is possible.

**Reply by obversepebble** [reply]
Yes, that's exactly what I am looking for. Is there an example that you can point to? I need web3auth to support both web2 and web3 logins – web2 logins powered by Firebase's authentication module (I've implemented this), and web3 logins creating users in firebase (I'm looking for an example).

**Comment by borsemayur2** [comment]
Hi,

I'm looking for similar solution.

I'm able to login with `signInWithGoogle` Firebase API and pass `idToken`  to `connectTo` Web3Auth API.
This creates user on firebase and web3auth.

Now, I want to login with Metamask wallet and create user on firebase and web3auth.
How can I achieve it?

Is there any way to integrate firebase with `@web3auth/metamask-adapter` ?


```JavaScript
const login = async () => {
    const loginRes = await signInWithGoogle(); // Need to use Metamask Login here
    const idToken = await loginRes.user.getIdToken(true);

    const web3authProvider = await web3auth.connectTo(
      WALLET_ADAPTERS.OPENLOGIN,  // Need to  use METAMASK here
      {
        loginProvider: "jwt",
        extraLoginOptions: {
          id_token: idToken,
          verifierIdField: "sub",
          domain: "http://localhost:3000",
        },
      }
    );
};
```

**Comment by ether-wan** [comment]
Hello ! 
Anyone successfully integrate metamask with firebase? 
Looking too for a solution

**Comment by anand-dehru** [comment]
Hello everyone,
I have also used moralis and it is working fine but I want toa achieve this with alternate method or you can say without moralis extension.

If anyone know or has any reference please tell.

**Reply by anand-dehru** [reply]
@obversepebble Have you achieved it without using moralis?
